HIP/hip-1340.md

```diff
@@ -0,0 +1,389 @@
+
+The second aspect of EOA Code Delegation is an update to the EVM call processing logic.
+
+When a call is made to an account - either a top-level EVM transaction call (i.e. when `transaction.to == EOA address`) or an inner call (e.g. `address(eoa).call(..)` call in a Smart Contract) - HSCS will:
+
+1. Resolve the target Account of the call
+2. Read `delegation_address`
+3. If `delegation_address` is non-empty: resolve the contract account that is the target of the delegation and run its EVM code in the context of the original EOA, subject to the limitations described below
+
+Throughout the EVM execution any queries for the code of an account (e.g. when using the `EXTCODECOPY` opcode) will be handled as follows:
+
+1. Resolve the target entity
+2. If it's a Contract: use `ContractStateStore` to fetch the bytecode
+3. If it's an Account: prefix the `delegation_address` with `0xef0100` to form an EVM-compatible Delegation Indicator and return it as the bytecode. If `delegation_address` is unset, then return an empty bytecode.
+
+One notable difference is that EVM code execution might result in EVM storage updates on the original EOA account as well as emit EVM events.
+As a result, all queries related to EVM storage and logs previously available for contract accounts should now also function for EOAs, which include:
+- `eth_getStorageAt` RPC method
+- `eth_getLogs` RPC method
+- all related APIs on the Mirror Node, including `/api/v1/contracts/*` endpoints (like `/api/v1/contracts/call`, `/api/v1/contracts/results/logs`, etc), which should be able to answer to any EOA requests
+
+In essence, all facilities available for contract accounts should also be availalbe for EOAs.
+
+Aside from code execution, the call might still carry a value transfer. The only difference in that regard is that if the code execution reverts, than the associated value transfer should also revert.
+
+The following checks will be applied before executing the EVM code of the target contract:
+
+- targetting a non-existent account results in a no-op code execution
+- targetting a precompile address results in a no-op code execution
+- targetting a system contract (HTS, HAS, etc) address results in a no-op code execution
+- targetting a system account address results in a no-op code execution
+- targetting an existing EOA with empty code results in a no-op code execution
+- targetting an existing EOA, which itself contains code delegation results in a no-op code execution
+- targetting an existing contract with non-empty code results in the execution of its EVM code, with the exception of calls matching one of Hedera Account Service proxy methods (described in detail below)
+
+In case Code Delegation points to another delegation, creating a potential chain or loop of delegations, only the first one will be resolved and the code of the target address will be interpreted literally, even if it's a Delegation Indicator.
```

Comment:
Wondering which of the following would happen in a scenario whereby EOA A delegates to EOA B, and EOA B delegates to Contract C?(EOA A → (delegation) → EOA B → (delegation) → Contract C)

Would calling EOA A do:

Option A: 
    1. Resolve delegation to EOA B
    2. Load EOA B's code (which is 0xef0100 || address_of_C)
    3. Execute 0xef0100...  as bytecode (which would fail since 0xef is banned)

Option B:
    1. Resolve delegation to EOA B
    2. See EOA B has a delegation
    3. No-op (per "targetting an existing EOA, which itself contains code delegation results in a no-op code execution")

I think Option B is what would happen but do we wanna explicitly clarify this in the HIP to avoid confusion?